src/installer.ts

```diff
@@ -0,0 +1,159 @@
+                console.log(`Minikube CLI (minikube) version returned: ${version}`);
+                resolve(version);
+                return;
+            }).catch ( (error) => {
+                console.log(`Minikube CLI (minikube)  unavailable: ${error}`);
+				reject(new Error(error));
+				return;
+        });
+    });
+}
+
+async function checkIfCLIAvailable(command : string) : Promise<string>{
+	return new Promise<string>( (resolve, reject) => {
+		let runCommand = child_process.exec(command);
+		if (runCommand.stdout) {
+			runCommand.stdout.on('data', function (data) {
+				let output : string = data as string;
+				resolve(output);
+				return;
+			});
+		}
+		if (runCommand.stderr) {
+			runCommand.stderr.on('data', function (data) {
+				let error : string = data as string;
+				reject(new Error(error));
+				return;
+			});
+		}
+	});
+}
+
+export async function installKamel(shell: Shell): Promise<Errorable<null>> {
+
+    const version = '1.0.0-M1'; //need to retrieve this if possible, but have a default
+    await checkKamelCLIVersion().then((currentVersion) => {
+        const currentVersionString = currentVersion as string;
+        if (version.toLowerCase() === currentVersionString.toLowerCase()) {
+            // no need to install, it's already here
+            extension.shareMessageInMainOutputChannel(`Apache Camel K CLI version ${currentVersionString} available`);
+            return { succeeded: true, result: null };
+        }
+    }).catch ( (error) => {
+        console.error(error);
+    });
+
+    const os = platform();
+    const isWindows = (os === 'win32');
+    const binFile = (!isWindows) ? kamel : kamel_windows;
+
+    const installFolder = getInstallFolder(shell, kamel);
+    mkdirp.sync(installFolder);
+
+    const kamelUrl = `https://github.com/apache/camel-k/releases/download/${version}/camel-k-client-${version}-${os}-64bit.tar.gz`;
+    const downloadFile = path.join(installFolder, binFile);
+
+    extension.shareMessageInMainOutputChannel(`Downloading kamel cli tool from ${kamelUrl} to ${downloadFile}`);
+
+    grabTarGzAndUnGZ(kamelUrl, installFolder).then( (flag) => {
+        console.log(`Downloaded ${downloadFile} successfully: ${flag}`);
+        try {
+            if (fs.existsSync(downloadFile)) {
+                if (shell.isUnix()) {
+                    fs.chmodSync(downloadFile, '0777');
+                }
+                config.addKamelPathToConfig(downloadFile);
+            }
+          } catch(err) {
+            console.error(err);
+            return { succeeded: false, error: [`Failed to download kamel: ${err}`] };
+          }
+    })
+    .catch ( (error) => {
+        console.log(error);
+        return { succeeded: false, error: [`Failed to download kamel: ${error}`] };
+    });
+    return { succeeded: true, result: null };
+}
+
+function getInstallFolder(shell: Shell, tool: string): string {
+    return path.join(shell.home(), `.vs-camelk/tools/${tool}`);
```

Comment:
it leads to install in `/home/apupier/.vs-camelk/tools/kamel/kamel`
I think we should use the VS Code Data Storage https://code.visualstudio.com/api/extension-capabilities/common-capabilities#data-storage.
The ExtensionContext.globalStoragePath seems to be the one that is relevant in our context. As Camel K files are not able to specifiy which version of Kamel to use.